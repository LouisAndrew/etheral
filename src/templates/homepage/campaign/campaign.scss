.wrap {
    a {
        width: 100%;
        text-decoration: none;
        transition: 0.2s;

        max-width: 480px;

        &:hover {
            filter: brightness(0.6);
        }

        @media (min-width: 48em) {
            width: 320px;
        }

        @media (min-width: 64em) {
            width: 48%;
            max-width: 48%;
        }
    }
}
