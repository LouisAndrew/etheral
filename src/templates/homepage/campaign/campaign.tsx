import React from 'react';
import { FixedObject } from 'gatsby-image';
import { Link } from '@reach/router';

import { Flex, Box } from 'rebass';

import Tile from 'components/tile';
import './campaign.scss';

export type Props = {
    campaignData: {
        url: string;
        campaignLink: string;
        campaignName: string;
        img: {
            sources: FixedObject | FixedObject[];
        };
    }[];
};

const Campaign: React.FC<Props> = ({ campaignData }) => (
    <Box variant="outerWrapper" my={[4, 4, 6]}>
        <Flex
            width="100%"
            flexWrap="wrap"
            className="wrap"
            justifyContent="space-evenly"
            alignItems="center"
            px={[4, 4, 0]}
        >
            {campaignData.map(campaign => (
                <Link key={campaign.campaignName} to={campaign.campaignLink}>
                    <Tile
                        tileOnText={campaign.campaignName.toUpperCase()}
                        img={campaign.img}
                        imgAlt={campaign.campaignName}
                        url={campaign.url}
                        overflow="hidden"
                        width="100%"
                        mb={[3, 3, 5]}
                        height={['90vw', '80vw', '48vw']}
                        maxHeight={['unset', 420, 'unset', 700, 1024]}
                    />
                </Link>
            ))}
        </Flex>
    </Box>
);

export { Campaign };
