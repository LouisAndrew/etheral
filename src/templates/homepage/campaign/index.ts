import { Campaign } from './campaign';

export default Campaign;
