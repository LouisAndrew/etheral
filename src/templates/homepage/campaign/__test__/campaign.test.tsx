import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

// import { render, cleanup } from '@testing-library/react'
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Campaign, Props } from '../campaign';

describe('Campaign element', () => {
    const props: Props = {
        campaignData: [],
    };

    const Element = <Campaign {...props} />;

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Element, div);
    });

    it('matches snapshot', () => {
        const run = true;

        if (run) {
            const tree = renderer.create(Element).toJSON();
            expect(tree).toMatchSnapshot();
        }
    });
});
