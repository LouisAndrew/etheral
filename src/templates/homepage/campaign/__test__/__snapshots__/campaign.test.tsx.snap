// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campaign element matches snapshot 1`] = `
<div
  className="css-1vv4hue"
>
  <div
    className="wrap css-c9wy9e"
  />
</div>
`;
